pandas/core/arrays/base.py

```diff
@@ -37,6 +48,8 @@
 
 _extension_array_shared_docs: Dict[str, str] = dict()
 
+ExtensionArrayT = TypeVar("ExtensionArrayT", bound="ExtensionArray")
```

Comment:
I might have asked before, but can you explain again why this is needed, and the `"ExtensionArray"` as string version doesn't work? 


(I think I mentioned this before, but IMO we should have better documentation about the specific patterns we are using in pandas in our contributing guides)

Reply:
> (I think I mentioned this before, but IMO we should have better documentation about the specific patterns we are using in pandas in our contributing guides)

these are not specific patterns to pandas. https://mypy.readthedocs.io/en/stable/generics.html#generic-functions and https://mypy.readthedocs.io/en/stable/generics.html#generic-methods-and-generic-self

Reply:
> I might have asked before, but can you explain again why this is needed, and the `"ExtensionArray"` as string version doesn't work?

I do recall explaining it in the past (obviously didn't do a good job) won't bother with a generic explanation again, and next time I come across a concrete example of why this is needed will cc you in.